modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/execution/MuleOperationExecutor.java

```diff
@@ -85,9 +87,17 @@ public void execute(ExecutionContext<ComponentModel> executionContext, ExecutorC
   }
 
   private Map<String, ?> buildOperationParameters(CoreEvent inputEvent, ExecutionContextAdapter<ComponentModel> ctx) {
-    return from(inputEvent)
+    Map<String, Object> parameters = new HashMap<>();
+    from(inputEvent)
         .getOperationExecutionParams(ctx.getComponent().getLocation(), inputEvent.getContext().getId())
-        .getParameters();
+        .getParameters()
+        .forEach((key, value) -> parameters.put(key, mapParameterValue(value)));
+    return parameters;
+  }
+
+  private static Object mapParameterValue(Object parameterValue) {
```

Comment:
I just didn't need it to access anything from the instance so I thought it would be better to make it static, so it has access to only the things it is supposed to use.
But if we have some different convention around this, I can change it.

Reply:
Changed it.